azerothcore/azerothcore-wotlk
# feat(Core/Spell): Implement ValidateSpellInfo

* cherry-pick from https://github.com/TrinityCore/TrinityCore/commit/2b5d7eef3a091e67c27e853d43ac0bebbfa67531 & https://github.com/TrinityCore/TrinityCore/commit/a0a158b5b851db7e2c16819ec89e913d914a3aba





## Changes Proposed:
-  Implement ValidateSpellInfo for bool Validate() 
```cpp
// This allows to use it this way, short and concise 
        bool Validate(SpellInfo const* /*spellInfo*/) override
        {
            return ValidateSpellInfo({ SPELL_ID });
        }

// Instead of this way
        bool Validate(SpellInfo const* /*spellInfo*/) override
        {
            if (!sSpellMgr->GetSpellInfo(SPELL_ID))
                return false;
            return true;
        }
```

## Tests Performed:
- Builds
- tested in-game

```
2021-01-22 00:05:52 Validating spell scripts...
2021-01-22 00:05:52 >> Validated 2218 scripts in 5 ms
```



## How to Test the Changes:

1. Test any spellscript, I personally tested it with `spell_item_instant_statue`
2. First test with `return ValidateSpellInfo({ SPELL_ID });` to see that it works
3. then test with `return !ValidateSpellInfo({ SPELL_ID });` and see that you will get this error while starting WorldServer as intended
```
ERROR: TSCR: Spell `75731` did not pass Validate() function of script `spell_item_instant_statue` - script will be not added to the spell
```

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
